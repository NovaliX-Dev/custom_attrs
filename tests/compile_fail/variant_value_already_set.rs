use custom_attrs::CustomAttrs;

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
enum Enum {
    #[attr(a = 5)]
    #[attr(a = 5)]
    Variant1,

    #[attr(
        a = 3,
        a = 3
    )]
    Variant2
}

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
enum Enum2 {
    #[attr(a = #self.0)]
    #[attr(a = #self.0)]
    Variant1(usize),

    #[attr(
        a = #self.field1,
        a = #self.field1
    )]
    Variant2 {
        field1: usize
    }
}


fn main() {
    // let _a = Enum::Variant1.get_a();
}