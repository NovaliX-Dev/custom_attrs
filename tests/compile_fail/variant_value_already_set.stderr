error: The value is already set for this attribute.
 --> tests/compile_fail/variant_value_already_set.rs:7:12
  |
7 |     #[attr(a = 5)]
  |            ^^^^^

error: First value of `a` is set here.
 --> tests/compile_fail/variant_value_already_set.rs:6:16
  |
6 |     #[attr(a = 5)]
  |                ^

error: The value is already set for this attribute.
  --> tests/compile_fail/variant_value_already_set.rs:12:9
   |
12 |         a = 3
   |         ^^^^^

error: First value of `a` is set here.
  --> tests/compile_fail/variant_value_already_set.rs:11:13
   |
11 |         a = 3,
   |             ^

error: The value is already set for this attribute.
  --> tests/compile_fail/variant_value_already_set.rs:21:12
   |
21 |     #[attr(a = #self.0)]
   |            ^^^^^^^^^^^

error: First value of `a` is set here.
  --> tests/compile_fail/variant_value_already_set.rs:20:16
   |
20 |     #[attr(a = #self.0)]
   |                ^^^^^^^^

error: The value is already set for this attribute.
  --> tests/compile_fail/variant_value_already_set.rs:26:9
   |
26 |         a = #self.field1
   |         ^^^^^^^^^^^^^^^^

error: First value of `a` is set here.
  --> tests/compile_fail/variant_value_already_set.rs:25:13
   |
25 |         a = #self.field1,
   |             ^^^^^^^^^^^^
